apps/passport-server/src/services/userService.ts

```diff
@@ -304,6 +307,60 @@ export class UserService {
     if (!user) res.status(404);
     res.json(user || null);
   }
+
+  public async handleNewDeviceLogin(
+    secret: string,
+    email: string,
+    commitment: string,
+    res: Response
+  ): Promise<void> {
+    try {
+      const ticket = await fetchDevconnectDeviceLoginTicket(
+        this.context.dbPool,
+        email,
+        secret
+      );
+
+      if (!ticket) {
+        const err = `Secret key is not valid, or no such device login exists.`;
+        res.status(500).send(err);
+        throw new Error(err);
```

Comment:
This throw is caught in the function that calls this one, where the `catch` block *also* sends a response (`res.sendStatus(500)`). This is not what you want to do - express throws an error in the case that you send multiple responses using the same `Response` object.